src/Riok.Mapperly/Descriptors/MappingBuilders/ConvertInstanceMethodMappingBuilder.cs

```diff
@@ -0,0 +1,55 @@
+using Microsoft.CodeAnalysis;
+using Riok.Mapperly.Abstractions;
+using Riok.Mapperly.Descriptors.Mappings;
+using Riok.Mapperly.Helpers;
+
+namespace Riok.Mapperly.Descriptors.MappingBuilders;
+
+public static class ConvertInstanceMethodMappingBuilder
+{
+    public static SourceObjectMethodMapping? TryBuildMapping(MappingBuilderContext ctx)
+    {
+        if (!ctx.IsConversionEnabled(MappingConversionType.ToTargetMethod))
+            return null;
+
+        var targetIsNullable = ctx.Target.NonNullable(out var nonNullableTarget);
+
+        // ignore `ToString` mapping for backward compatibility
+        if (nonNullableTarget.SpecialType == SpecialType.System_String)
+            return null;
+
+        var methodName = GetMappingMethodName(ctx);
+
+        var methodCandidates = ctx
+            .SymbolAccessor.GetAllMethods(ctx.Source, methodName)
+            .Where(x => x is { IsStatic: false, ReturnsVoid: false, IsAsync: false, Parameters.Length: 0 })
+            .ToArray();
```

Comment:
The only reason I chose `ToArray` is because the list is not mutable and should not be mutable further down the code.
I'll change it to `ToList` since it works better.